docs/upgrade.rst

```diff
@@ -8,6 +8,22 @@ to a new version. But it goes into more detail. This includes things like
 deprecated APIs and their replacements, build system changes, general code
 modernization and other useful information.
 
+.. _upgrade-guide-2.10:
+
+v2.10
+=====
+
+The current scoped GIL implementation doesn't support nested access. In pybind11
```

Comment:
Please pick and choose:

The scoped GIL management implementations in previous pybind11 releases do not support nested access. In pybind11 2.10.1, a configuration option ``PYBIND11_SIMPLE_GIL_MANAGEMENT`` was added, defaulting to ``OFF``; the simpler implementations support nested access, but do not support dissociation (the ``true`` parameter of ``gil_scope_release``).  In pybind11 2.11, we plan to change the default to ``ON``. We also plan to have an example for implementing dissociation without direct support from pybind11.